extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -484,6 +488,17 @@ public SupervisorReport getStatus()
     return generateReport(true);
   }
 
+  @Override
+  public Map<String, Object> getStats()
+  {
+    try {
+      return getCurrentTotalStats();
+    }
+    catch (Exception e) {
```

Comment:
does this need to catch all exceptions?

Reply:
`InterruptedException` is not handled properly here

Reply:
Added handler for InterruptedException.

For other exceptions, changed this to log an error and return an empty map.